modules/subscriptions.php

```diff
@@ -137,31 +153,65 @@ function subscription_post_page_metabox() {
 		<?php endif;
 	}
 
-	/*
-	 * Disable Subscribe on Single Post
-	 * Save the meta
+	/**
+	 * Checks whether or not the post should be emailed to subscribers
 	 *
-	 * @return post object, post ID, or false if error
+	 * It checks for the following things in order:
+	 * - Usage of filter jetpack_subscriptions_exclude_these_categories
+	 * - Usage of filter jetpack_subscriptions_include_only_these_categories
+	 * - Existence of the per-post checkbox option
+	 *
+	 * Only one of these can be used at any given time.
+	 *
+	 * @param $new_status string - the "new" post status of the transition when saved
+	 * @param $old_status string - the "old" post status of the transition when saved
+	 * @param $post obj - The post object
 	 */
-	function save_subscribe_meta(){
-		global $post;
-		if ( ! is_object( $post ) ) {
-			return false;
-		}
-		// Check nonce
-		if ( empty( $_POST['disable_subscribe_nonce'] ) || ! wp_verify_nonce( $_POST['disable_subscribe_nonce'], 'disable_subscribe' ) ) {
-			return false;
+	function maybe_send_subscription_email( $new_status, $old_status, $post ) {
+		// Only do things on publish
+		if ( 'publish' !== $new_status ) {
+			return;
 		}
 		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
-			return $post->ID;
+			return;
```

Comment:
Nitpick mode: if we're returning nothing here, perhaps we can join this with the previous `if`?